scripts/generate_learning_path_markdown.js

```diff
@@ -1,127 +1,57 @@
+const getContributors = require('./get_contributors')
+const mkdirSync = require('./mkdir_sync')
+const getArticleFiles = require('./get_article_files')
 
-  const mkdirSync = (dir) => {
-    try {
-      fs.mkdirSync(dir)
-    } catch (e) {
-      if (e.code !== 'EEXIST') {
-        console.log(e)
-      }
-    }
-  }
-
-  const getArticleFiles = (path) => {
-    return fs.readdirSync(path).reduce((articles, filename) => {
-      const filePath = `${path}/${filename}`
-      if (filePath.match(/\d\d/) && !filePath.includes('-script.asciidoc')) {
-        return [...articles, {
-          filePath,
-          asciiDoc: fs.readFileSync(filePath, 'utf-8')
-        }]
-      } else {
-        return articles
-      }
-    }, [])
-  }
-
-  const writeMarkdownFile = (filePath, frontMatter) => {
-    const frontMatterTerminator = '---'
-    const originStatement = '<!--- This file autogenerated from https://github.com/InnerSourceCommons/InnerSourceLearningPath/blob/master/scripts/generate_learning_path_markdown.js -->'
-    const output = [frontMatterTerminator, YAML.stringify(frontMatter).trim(), frontMatterTerminator, originStatement].join(EOL)
-    fs.writeFileSync(filePath, output)
-  }
+const sections = require('./section_data.json')
 
-  const sections = [
-    {
-      learning_path_group: 'Introduction',
-      dirName: 'introduction',
-      workbook: '01-introduction.asciidoc',
-      translations: ['de', 'it', 'ja', 'zh', 'ru'],
-      renderArticles: true
-    },
-    {
-      learning_path_group: 'Trusted Committer',
-      dirName: 'trusted-committer',
-      workbook: '02-trusted-committer.asciidoc',
-      translations: ['de', 'zh'],
-      renderArticles: true
-    },
-    {
-      learning_path_group: 'Contributor',
-      dirName: 'contributor',
-      workbook: '04-contributor.asciidoc',
-      translations: ['it', 'ja', 'zh'],
-      renderArticles: true
-    },
-    {
-      learning_path_group: 'Product Owner',
-      dirName: 'product-owner',
-      workbook: '03-product-owner.asciidoc',
-      translations: ['zh'],
-      renderArticles: true
-    },
-  ]
+module.exports = async (writeDir, generatorFn, workbookFn, createTranslationFolder) => {
+  mkdirSync(join('.', writeDir))
 
-  mkdirSync('./learningpath')
-
-  sections.forEach(({ learning_path_group, dirName, workbook, translations, renderArticles }) => {
-    const baseReadPath = `../${dirName}`
-    const baseWritePath = `./learningpath/${dirName}`
+  sections.forEach(section => {
+    const { dirName, translations } = section
+    const baseReadPath = join('..', dirName)
+    const baseWritePath = join('.', writeDir, dirName)
     mkdirSync(baseWritePath)
 
     translations.concat('' /* The English original */).forEach(async (translation) => {
       const isTranslation = translation !== ''
       const writePath = join(baseWritePath, translation)
-      mkdirSync(writePath)
+      if (createTranslationFolder) mkdirSync(writePath)
```

Comment:
I don't understand why this folder is sometimes needed and sometimes not?

Reply:
innersourcecommons.net doesn't use separate folders for translations. They appear in the same folder but with a special suffix (see https://github.com/InnerSourceCommons/innersourcecommons.net/pull/52).